pkg/endpoints/pipeline.go

```diff
@@ -403,8 +555,42 @@ func getDateTimeAsString() string {
 	return strconv.FormatInt(time.Now().Unix(), 10)
 }
 
+// defines and creates a resource of a specifed type and returns a pipeline resource reference for this resource
+// takes a manual pipeline run data struct, namespace for the resource creation, resource name to refer to it and the resource type
+func (r Resource) createPipelineResource(resourceData ManualPipelineRun, namespace, resourceName string, resourceType v1alpha1.PipelineResourceType) v1alpha1.PipelineResourceRef {
+	logging.Log.Debug("Creating PipelineResource of type %s", resourceType)
+
+	startTime := getDateTimeAsString()
+
+	registryURL := resourceData.REGISTRYLOCATION
+	resourceName = fmt.Sprintf("%s-%s", resourceName, startTime)
+
+	var paramsForResource []v1alpha1.Param
+	var typeToUse v1alpha1.PipelineResourceType
+	// Unique names are required so timestamp them.
+	if resourceType == v1alpha1.PipelineResourceTypeGit {
+		typeToUse = resourceType
+		paramsForResource = []v1alpha1.Param{{Name: "revision", Value: resourceData.GITCOMMIT}, {Name: "url", Value: resourceData.REPOURL}}
+	} else if resourceType == v1alpha1.PipelineResourceTypeImage {
+		urlToUse := fmt.Sprintf("%s/%s:%s", registryURL, strings.ToLower(resourceData.REPONAME), resourceData.GITCOMMIT)
+		typeToUse = resourceType
```

Comment:
Please remove the lines `typeToUse = resourceType` and `var typeToUse v1alpha1.PipelineResourceType`, because it's just changing the name of the variable `resourceType` to `typeToUse` in both cases of this `if` statement. If the variable should be named `typeToUse`, then change it in the function arguments list.